pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -137,14 +141,24 @@ def submit(
         configuration = self._convert_configuration(
             config=kwargs.get("config", None), emulator=emulator
         )
-        create_batch_fn = backoff_decorator(self._sdk_connection.create_batch)
-        batch = create_batch_fn(
-            serialized_sequence=sequence.to_abstract_repr(),
-            jobs=job_params or [],  # type: ignore[arg-type]
-            emulator=emulator,
-            configuration=configuration,
-            wait=wait,
-        )
+        batch = None
+        if submission_id:
+            submit_jobs_fn = backoff_decorator(self._sdk_connection.add_jobs)
+            batch = submit_jobs_fn(
+                batch_id=submission_id,
+                jobs=job_params or [],  # type: ignore[arg-type]
+            )
```

Comment:
Another thing here: what makes more sense within the backend interface is that the `RemoteResults()` returned when adding jobs to a batch contains only the added jobs, not the full batch. This will require changing `RemoteResults` to accept the  job IDs for the subset of jobs in a batch it should care about.

Reply:
This still needs to be addressed. I can take over that part if you wish